CMakeLists.txt

```diff
@@ -19,175 +14,42 @@ else (ENABLE_DEBUG_GRAPHICS)
-catkin_package(
-        INCLUDE_DIRS
-        include
-        ${protogen_include_dirs}
-        LIBRARIES
-        roboteam_utils
-        ${PROJECT_NAME}_proto
-)
-
-include_directories(
-        ${catkin_INCLUDE_DIRS}
-        include
-        ${PROTOBUF_INCLUDE_DIR}
-        ${PROTOBUF_INCLUDE_DIR}/../../
-)
-
-# Create lists of files to be generated.
-set(proto_gen_cpp_files "")
-set(proto_gen_python_files "")
-foreach (proto_file ${proto_files})
-    get_filename_component(proto_name ${proto_file} NAME_WE)
-    list(APPEND proto_gen_cpp_files ${proto_gen_cpp_dir}/${proto_name}.pb.h ${proto_gen_cpp_dir}/${proto_name}.pb.cc)
-    list(APPEND proto_gen_python_files ${proto_gen_python_dir}/${proto_name}_pb2.py)
-endforeach (proto_file ${proto_files})
-
-# Run protoc and generate language-specific headers.
-add_custom_command(
-        OUTPUT ${proto_gen_cpp_files} ${proto_gen_python_files}
-        COMMAND ${PROTOBUF_PROTOC_EXECUTABLE} --proto_path=${proto_dir} --cpp_out=${proto_gen_cpp_dir} --python_out=${proto_gen_python_dir} ${proto_files}
-        DEPENDS ${PROTOBUF_PROTOC_EXECUTABLE} ${proto_files}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-)
-
-set_source_files_properties(${proto_gen_cpp_files} PROPERTIES GENERATED TRUE)
-set_source_files_properties(${proto_gen_python_files} PROPERTIES GENERATED TRUE)
-
-# Make the python message folder a python package.
-file(APPEND ${proto_gen_python_dir}/__init__.py "")
-
-add_custom_target(${PROJECT_NAME}_generate_headers
-        DEPENDS ${proto_gen_cpp_files} ${proto_gen_python_files}
-        )
-
-# Create proto library for linking.
-add_library(${PROJECT_NAME}_proto ${proto_gen_cpp_files})
-target_link_libraries(${PROJECT_NAME}_proto ${PROTOBUF_LIBRARY})
-add_dependencies(${PROJECT_NAME}_proto ${PROJECT_NAME}_generate_headers)
-
-install(TARGETS ${PROJECT_NAME}_proto
-        ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-        )
-
-install(DIRECTORY ${proto_gen_cpp_dir}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-        FILES_MATCHING PATTERN "*.h"
-        )
-
-install(DIRECTORY ${proto_gen_python_dir}/
-        DESTINATION ${CATKIN_PACKAGE_PYTHON_DESTINATION}
-        FILES_MATCHING PATTERN "*.py"
-        )
-
-##########################
-## Constants generation ##
-##########################
 
-set(RTT_CONSTANTS_HEADER_FILE "${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}/constants.h")
-set(RTT_CONSTANTS_SOURCE_FILE "${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_LIB_DESTINATION}/roboteam_utils/constants.cpp")
-configure_file(include/roboteam_utils/constants.h ${RTT_CONSTANTS_HEADER_FILE} COPYONLY)
-configure_file(src/constants.cpp ${RTT_CONSTANTS_SOURCE_FILE} COPYONLY)
 
-###########################
-## Regular library stuff ##
-###########################
 
-add_library(roboteam_utils
+add_library(roboteam_utils STATIC
```

Comment:
good. This is necessary for runtime performance but will drag compile times down.